router/utils/utils.go

```diff
@@ -45,7 +45,7 @@ func loadConfig() {
 func getRetentionTimeForDestination(destID string) time.Duration {
 	destJobRetentionFound := config.IsSet("Router." + destID + ".jobRetention")
 	if destJobRetentionFound {
-		return config.GetDuration("Router"+"."+destID+"jobRetention", 720, time.Hour)
+		return config.GetDuration("Router."+destID+".jobRetention", 720, time.Hour)
```

Comment:
@chandumlg are we really using this configuration option in production? How can this be?

Reply:
We use this for open source people so that if for some reason their server goes into degraded mode then upon resumption of normal mode we don’t loose their jobs. We set it to 24hr in our production setup through operator

Reply:
Are we setting it properly (with the dot), improperly (without the dot) or both ways?

Reply:
I’m sorry I didn’t understand what you are trying to say. Can you elaborate a bit?

Reply:
https://github.com/rudderlabs/rudderstack-operator/blob/master/operator-helm/valuefiles/values.yaml#L121

This is what we set in production environment

Reply:
@cisse21  I am referring to the per-destinationId configuration override.
The current code in `master` as you can see here is testing the existence of `"Router." + destID + ".jobRetention"` and then reads the value from `"Router"+"."+destID+"jobRetention"`!

Reply:
Got It…. We set it at global level which is not without any destination Id flag. We can override the default behaviour for destination Id by setting appropriate flag per customer

Reply:
@atzoum This is a bug. Thanks for fixing this. 👍